x-pack/plugins/ml/public/application/components/create_data_view_button/create_data_view_button.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButton } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React, { Fragment, useCallback, useRef } from 'react';
+import { useMlKibana } from '../../contexts/kibana';
+
+export const CreateDataViewButton = ({
+  onDataViewCreated,
+  allowAdHocDataView = false,
+}: {
+  onDataViewCreated: (id: string, type: string, name?: string) => void;
+  allowAdHocDataView?: boolean;
+}) => {
+  const { dataViewEditor } = useMlKibana().services;
+  const canEditDataView = Boolean(dataViewEditor?.userPermissions.editDataView());
+  const closeDataViewEditor = useRef<() => void | undefined>();
```

Comment:
This is currently not being used. 
It should be used to close the editor when unmounting the component.

Reply:
Updated here [`2262a7f` (#166668)](https://github.com/elastic/kibana/pull/166668/commits/2262a7fafdfe9e9c8d08e35efcab65e150eeba15)